great_expectations/data_context/migrator/cloud_migrator.py

```diff
@@ -297,16 +299,14 @@ def _send_validation_results(
         serialized_validation_results: Dict[str, dict],
         test_migrate: bool,
     ) -> None:
-        print("[Step 4/4]: Sending validation results]")
-        if test_migrate:
-            return
-
+        print("[Step 4/4]: Sending validation results")
         self._process_validation_results(
-            serialized_validation_results=serialized_validation_results
+            serialized_validation_results=serialized_validation_results,
+            test_migrate=test_migrate,
         )
 
     def _process_validation_results(
-        self, serialized_validation_results: Dict[str, dict]
+        self, serialized_validation_results: Dict[str, dict], test_migrate: bool
```

Comment:
You could use dependency injection here instead of this `test_migrate` flag.

Like a `results_processor` callable (which in prod would be this `_post_to_cloud_backend` method), but could be something else for tests or other scenarios.

Another option would just be to remove this `test_migrate` bool and make `self._post_to_cloud_backend` a spy for your tests. Would that solve the issue?
https://pytest-mock.readthedocs.io/en/latest/usage.html#spy